### PR TITLE
MVP_openapi_implementation

### PR DESCRIPTION
**Pull request by zhdanov-nikolay**
Added module for OpenApi generation documentation.

Added:
- documentation endpoint
- util classes
- test classes
- java docs
- readme file

### CODE REVIEW

**Comment by vdiravka** [comment]
Hi @zhdanov-nikolay 

Please add JavaDocs for `OpenapiWS`, `ApiInfo` and create README.md with module description.
Since this PR has several todo: `Needs to be done: necessary to add security tags, do refactoring, move some logic to Util class` please convert it to draft.

**Comment by zhdanov-nikolay** [comment]
Hi @vdiravka. All todo are resolved, added javadoc and readme file.

File: oap-ws-openapi-ws/src/test/resources/META-INF/oap-module.conf [hunk 1]
```diff
@@ -0,0 +1,8 @@
+name = oap-ws-openapi-ws-test
+dependsOn = oap-ws-openapi-ws
+services {
+  test-service {
```
Comment (vdiravka) [inline oap-ws-openapi-ws/src/test/resources/META-INF/oap-module.conf:4]: usually we are not specifying test classes as oap services. Check `oap.ws.api.ApiWSTest` for an example

**Review by vdiravka — CHANGES_REQUESTED** [review]
Thanks for the contribution. It enriches current system/api functionality. 
Could you add some comparison with an original `system/api` `ApiWS`, possibly pros and cons etc?

File: oap-ws-openapi-ws/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
+# OAP-WS Openapi module
+Generating OpenApi documentation for OAP web services
+
+
+## General Information 
+- Generates OpenApi (version 3.0+) documentation for all classes which are used as web services in an easy way.
```
[the remark this answers is not in the document]
Comment (zhdanov-nikolay) [reply oap-ws-openapi-ws/README.md:6]: As far as I know there are only 3 java OAS implementations: swagger, KaiZen-OpenApi-Parser and spring-openapi(which uses swagger under the hood). Swagger uses the newest (from all others) OAS version and recieves frequent updates, so I decided to use it as an implementation. I think it'll take much more time and effort to support own implementation.

File: oap-ws-openapi-ws/src/test/resources/META-INF/oap-module.conf (hunk 1 above)
Comment (zhdanov-nikolay) [reply oap-ws-openapi-ws/src/test/resources/META-INF/oap-module.conf:4]: I used oap-ws-api module [configuration ](https://github.com/oaplatform/oap-ws/blob/master/oap-ws-api-ws/src/test/resources/META-INF/oap-module.conf) as an example for testing.

Comment (vdiravka) [reply oap-ws-openapi-ws/src/test/resources/META-INF/oap-module.conf:4]: I see now, it is just web service. Do you want to add Test for `TestWS`? Ot the aim was just to use it for docs?

Comment (zhdanov-nikolay) [reply oap-ws-openapi-ws/src/test/resources/META-INF/oap-module.conf:4]: Yes, the main goal is to describe `TestWS` web service via OpenApi. There are no specific logic in this class which should be covered with tests.

File: oap-ws-openapi-ws/src/test/resources/oap/ws/openapi/OpenapiWSTest/openapi.json [hunk 3]
```diff
@@ -0,0 +1,677 @@
+{
+  "openapi": "3.0.1",
```
[the remark this answers is not in the document]
Comment (zhdanov-nikolay) [reply oap-ws-openapi-ws/src/test/resources/oap/ws/openapi/OpenapiWSTest/openapi.json]: Moved version to 3.0.3

**Review by vdiravka — APPROVED** [review]
+1 LGTM

**Pull request merged**